src/EmergencyDropSpell.sol

```diff
@@ -0,0 +1,106 @@
+}
+
+interface DsPauseLike {
+    function plans(bytes32) external view returns (bool);
+    function drop(address, bytes32, bytes calldata, uint256) external;
+}
+
+interface EmergencySpellLike {
+    function action() external view returns (address);
+    function cast() external;
+    function description() external view returns (string memory);
+    function done() external view returns (bool);
+    function eta() external view returns (uint256);
+    function expiration() external view returns (uint256);
+    function log() external view returns (address);
+    function nextCastTime() external view returns (uint256);
+    function officeHours() external view returns (bool);
+    function pause() external view returns (DsPauseLike);
+    function schedule() external;
+    function sig() external view returns (bytes memory);
+    function tag() external view returns (bytes32);
+}
+
+/// @title A spell that drops a plan from `MCD_PAUSE` when is cast.
+contract EmergencyDropSpell is EmergencySpellLike {
+    bool public constant officeHours = false;
+    uint256 public constant expiration = type(uint256).max;
+    address public constant log = address(0);
+
+    /// @notice The Protego factory that deployed the spell.
+    ProtegoLike public immutable protego;
+    /// @notice The MCD_PAUSE instance.
+    DsPauseLike public immutable pause;
+    /// @notice The original spell action address.
+    address public immutable action;
+    /// @notice The original spell action tag (i.e.: `extcodehash`).
+    bytes32 public immutable tag;
+    /// @notice The original spell expiry time.
+    uint256 public immutable eta;
+    /// @notice The original spell encoded call.
+    bytes public sig;
+
+    /// @param _protego The Protego factory that deployed the spell.
+    /// @param _pause The MCD_PAUSE instance.
+    /// @param _usr The original spell action address.
+    /// @param _tag The original spell action tag (i.e.: `extcodehash`).
+    /// @param _fax The original spell encoded call.
+    /// @param _eta The original spell expiry time.
+    constructor(address _protego, address _pause, address _usr, bytes32 _tag, bytes memory _fax, uint256 _eta) {
+        protego = ProtegoLike(_protego);
+        pause = DsPauseLike(_pause);
+        action = _usr;
+        tag = _tag;
+        sig = _fax;
+        eta = _eta;
+    }
+
+    /// @notice Returns the description of the spell in the format "MakerDAO Drop Spell: <ID>"
+    function description() external view returns (string memory) {
+        return string(abi.encodePacked("MakerDAO Drop Spell: ", protego.id(action, tag, sig, eta)));
+    }
+
+    /// @notice Returns whether the original spell has been planned or not.
+    function planned() public view returns (bool) {
+        return protego.planned(action, tag, sig, eta);
+    }
+
+    /// @notice Drops the original spell.
+    function schedule() external {
+        pause.drop(action, tag, sig, eta);
+    }
```

Comment:
Should we also emit `Drop` event here for consistency (since it can be executed multiple times)? Or maybe even call protego.drop so that events accumulate under the same address

Reply:
`protego.drop` is only possible if `Protego` has the hat.

Reply:
Please check now. I think I found a good solution for the naming issue.

Reply:
> `protego.drop` is only possible if `Protego` has the hat. Also not sure if we should call this event `Drop` or `Schedule` 😅

You can do delegatecall, but I don't know if it's a good practice to use delegate call for the sake of having events in one place.

Reply:
Yeah, in this case it would work because there's no state involved, however I believe this would not land well with the existing security practices. A spell with `delegatecall` would cause an aneurysm in some people 😅.